source/elements/oneMKL/source/domains/blas/gemm_batch.rst

```diff
@@ -461,6 +465,181 @@ in ``a``, ``b`` and ``c`` are given by the ``batch_size`` parameter.
 
    Output event to wait on to ensure computation is complete.
 
+**Span API**
+
+.. rubric:: Description
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::column_major {
+       sycl::event gemm_batch(sycl::queue &queue,
+                              const sycl::span<onemkl::transpose> &transa,
+                              const sycl::span<onemkl::transpose> &transb,
+                              const sycl::span<std::int64_t> &m,
+                              const sycl::span<std::int64_t> &n,
+                              const sycl::span<std::int64_t> &k,
+                              const sycl::span<T> &alpha,
+                              const sycl::span<T*> &a,
+                              const sycl::span<std::int64_t> &lda,
+                              const sycl::span<T*> &b,
+                              const sycl::span<std::int64_t> &ldb,
+                              const sycl::span<T> &beta,
+                              sycl::span<T*> &c,
+                              const sycl::span<std::int64_t> &ldc,
+                              size_t num_groups,
+                              const sycl::span<size_t> &group_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+.. code-block:: cpp
+
+   namespace oneapi::mkl::blas::row_major {
+       sycl::event gemm_batch(sycl::queue &queue,
+                              const sycl::span<onemkl::transpose> &transa,
+                              const sycl::span<onemkl::transpose> &transb,
+                              const sycl::span<std::int64_t> &m,
+                              const sycl::span<std::int64_t> &n,
+                              const sycl::span<std::int64_t> &k,
+                              const sycl::span<T> &alpha,
+                              const sycl::span<T*> &a,
+                              const sycl::span<std::int64_t> &lda,
+                              const sycl::span<T*> &b,
+                              const sycl::span<std::int64_t> &ldb,
+                              const sycl::span<T> &beta,
+                              sycl::span<T*> &c,
+                              const sycl::span<std::int64_t> &ldc,
+                              size_t num_groups,
+                              const sycl::span<size_t> &group_size,
+                              const std::vector<sycl::event> &dependencies = {})
+   }
+
+.. container:: section
+
+   .. rubric:: Input Parameters
+
+   queue
+      The queue where the routine should be executed.
+
+   transa
```

Comment:
For all span input we should list somewhere what are the requirement on the span size.
Only 1, group_count and total_batch_size makes sense so all other sizes would raise an error.
For the output C only total_batch_size makes sense.

Reply:
Now included in the opening of the **gemm_batch (USM Version)** section